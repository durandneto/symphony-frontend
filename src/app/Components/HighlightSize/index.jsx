import React from 'react'
import PropTypes from 'prop-types'
import Container, { Row, Column } from '../Grid'
import HeaderFont from '../Typography/Header'
import BodyFont from '../Typography/Body'

const HighlightSizeComponent = props => (
  <Container xs loading={props.loading}>
    <Row justifyCenter>
      <Column>
        <HeaderFont h4 color={"Blue-Grey-300"} upperCase label={props.title} />
      </Column>
    </Row>
    <Row spaceAround autoColumn>
      <Column full >
        <Row justifyCenter>
          <Column>
            <HeaderFont h2 color={"Grey-800"} upperCase label={props.left.value} />
          </Column>
        </Row>
        <Row justifyCenter>
          <Column>
            <BodyFont sm color={"Blue-Grey-300"} upperCase label={props.left.label} />
          </Column>
        </Row>
      </Column>
      <Column full>
        <Row justifyCenter>
          <Column>
            <HeaderFont h2 color={"Grey-800"} upperCase label={props.right.value} />
          </Column>
        </Row>
        <Row justifyCenter>
          <Column>
            <BodyFont sm color={"Blue-Grey-300"} upperCase label={props.right.label} />
          </Column>
        </Row>
      </Column>
    </Row>
  </Container>
)

HighlightSizeComponent.defaultProps = {}

HighlightSizeComponent.propTypes = {
  title: PropTypes.string.isRequired,
  left: PropTypes.shape({
    value: PropTypes.number.isRequired,
    label: PropTypes.oneOfType([
        PropTypes.string,
        PropTypes.number,
      ]),
  }).isRequired,
  right: PropTypes.shape({
    value: PropTypes.number.isRequired,
    label: PropTypes.oneOfType([
        PropTypes.string,
        PropTypes.number,
      ]),
  }).isRequired,
  data: PropTypes.object,
}

export default HighlightSizeComponent
