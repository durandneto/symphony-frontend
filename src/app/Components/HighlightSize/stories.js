import React from 'react'
import { action } from '@storybook/addon-actions'
import Container, { Row, Column } from '../Grid'
import HighlightSize from '.'


export default {
  title: 'HIGHLIGHT|size/default',
}

const objHighlightSize = {
  title: "bundle size",
  left: {
    value: 10,
    label: "MINIFIED"
  },
  right: {
    value: 20,
    label: "MINIFIED + GZIPPED"
  }
}

export const Single = () => (
  <HighlightSize
    {...objHighlightSize}
  />
)

export const Double = () => (
  <Container>
    <Row justifyCenter autoColumn>
      <Column full>
        <HighlightSize
          {...objHighlightSize}
        />
      </Column>
      <Column full>
        <HighlightSize
          {...objHighlightSize}
        />
      </Column>
    </Row>
  </Container>
)
