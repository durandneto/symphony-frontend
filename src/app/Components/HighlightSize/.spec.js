import React from 'react'
import renderer from 'react-test-renderer'
import HighlightSize from '.'

const objHighlightSize = {
  title: "bundle size",
  left: {
    value: 10,
    label: "MINIFIED"
  },
  right: {
    value: 20,
    label: "MINIFIED + GZIPPED"
  }
}

describe('HighlightSize', () => {
  it ('Rendering HighlightSize without crashing', () => {
    const result = renderer.create(
      <HighlightSize
        {...objHighlightSize}
      />
    ).toJSON()
    expect(result).toMatchSnapshot()
  })
})
