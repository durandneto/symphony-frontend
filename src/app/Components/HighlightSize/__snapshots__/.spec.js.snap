// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HighlightSize Rendering HighlightSize without crashing 1`] = `
<div
  className="sc-AxjAm gSJRdo"
>
  <div
    className="sc-AxjAm sc-AxirZ gurTkc"
  >
    <div
      className="sc-AxjAm sc-AxiKw eXQwvW"
    >
      <h4
        className="sc-AxheI DjEpp"
        color="Blue-Grey-300"
        label="bundle size"
      >
        bundle size
      </h4>
    </div>
  </div>
  <div
    className="sc-AxjAm sc-AxirZ dSHUZK"
  >
    <div
      className="sc-AxjAm sc-AxiKw gVOZNl"
    >
      <div
        className="sc-AxjAm sc-AxirZ gurTkc"
      >
        <div
          className="sc-AxjAm sc-AxiKw eXQwvW"
        >
          <h2
            className="sc-AxhUy gTTLCL"
            color="Grey-800"
            label={10}
          >
            10
          </h2>
        </div>
      </div>
      <div
        className="sc-AxjAm sc-AxirZ gurTkc"
      >
        <div
          className="sc-AxjAm sc-AxiKw eXQwvW"
        >
          <span
            className="sc-fzozJi gHnIcl"
            color="Blue-Grey-300"
            label="MINIFIED"
          >
            MINIFIED
          </span>
        </div>
      </div>
    </div>
    <div
      className="sc-AxjAm sc-AxiKw gVOZNl"
    >
      <div
        className="sc-AxjAm sc-AxirZ gurTkc"
      >
        <div
          className="sc-AxjAm sc-AxiKw eXQwvW"
        >
          <h2
            className="sc-AxhUy gTTLCL"
            color="Grey-800"
            label={20}
          >
            20
          </h2>
        </div>
      </div>
      <div
        className="sc-AxjAm sc-AxirZ gurTkc"
      >
        <div
          className="sc-AxjAm sc-AxiKw eXQwvW"
        >
          <span
            className="sc-fzozJi gHnIcl"
            color="Blue-Grey-300"
            label="MINIFIED + GZIPPED"
          >
            MINIFIED + GZIPPED
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
